### PR TITLE
ADCM-3061

### PR DESCRIPTION
**Pull request by kuhella**

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Hey, @kuhella! Build has been launched!

File: python/api/cluster/serializers.py [hunk 1]
```diff
@@ -94,6 +94,32 @@ def update(self, instance, validated_data):
         return instance
 
 
+class ClusterUISerializer(ClusterSerializer):
+    action = CommonAPIURL(view_name='object-action')
+    edition = CharField(read_only=True)
+    prototype_version = SerializerMethodField()
+    prototype_name = SerializerMethodField()
+    prototype_display_name = SerializerMethodField()
+    upgrade = hlink('cluster-upgrade', 'id', 'cluster_id')
```
Comment (winmasta) [inline python/api/cluster/serializers.py:103]: `hlink` function seems to be redundant and add implicitness

File: python/api/cluster/serializers.py [hunk 2]
```diff
@@ -94,6 +94,32 @@ def update(self, instance, validated_data):
...
+    upgradable = SerializerMethodField()
+    get_upgradable = get_upgradable_func
```
Comment (winmasta) [inline python/api/cluster/serializers.py:105]: Also implicit behaviour

File: python/api/cluster/serializers.py [hunk 3]
```diff
@@ -94,6 +94,32 @@ def update(self, instance, validated_data):
...
+    concerns = ConcernItemUISerializer(many=True, read_only=True)
+    locked = BooleanField(read_only=True)
+    status = SerializerMethodField()
+
+    def get_prototype_version(self, obj):
```
Comment (winmasta) [inline python/api/cluster/serializers.py:110]: Type annotations can be added

File: python/api/service/serializers.py [hunk 4]
```diff
@@ -52,6 +52,21 @@ def create(self, validated_data):
...
+    version = serializers.SerializerMethodField()
+    concerns = ConcernItemUISerializer(many=True, read_only=True)
+    locked = serializers.BooleanField(read_only=True)
+    status = serializers.SerializerMethodField()
+
+    def get_version(self, obj):
+        return obj.prototype.version
+
+    def get_status(self, obj):
+        return status_api.get_service_status(obj)
```
Comment (winmasta) [inline python/api/service/serializers.py:67]: Explicit import is better `from cm.status_api import get_service_status`

File: python/cm/upgrade.py [hunk 5]
```diff
@@ -270,7 +270,7 @@ def rpm_cmp_reverse(obj1, obj2):
         if upg.upgradable:
             res.append(upg)
 
-    cm.issue.update_hierarchy_issues(obj)
+    # cm.issue.update_hierarchy_issues(obj)
```
Comment (winmasta) [inline python/cm/upgrade.py:273]: Commented out code can be deleted and shouldn't be pushed

Comment (kuhella) [reply python/cm/upgrade.py]: fixed

File: python/api/service/serializers.py (hunk 4 above)
Comment (kuhella) [reply python/api/service/serializers.py]: fixed

File: python/api/cluster/serializers.py (hunk 3 above)
Comment (kuhella) [reply python/api/cluster/serializers.py]: fixed

File: python/api/cluster/serializers.py (hunk 2 above)
Comment (kuhella) [reply python/api/cluster/serializers.py]: fixed

File: python/api/cluster/serializers.py (hunk 1 above)
Comment (kuhella) [reply python/api/cluster/serializers.py]: fixed

**Comment by kuhella** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @kuhella! Build has been launched!

**Review by winmasta — DISMISSED** [review]

**Comment by kuhella** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @kuhella! Build has been launched!

**Review by winmasta — APPROVED** [review]

**Pull request merged**